src/pages/home/report/ReportActionsView.js

```diff
@@ -110,6 +101,16 @@ class ReportActionsView extends React.Component {
     }
 
     componentDidUpdate(prevProps) {
+        // When the last action changes, wait three seconds, then record the max action
+        // If the report is already open, record the max action immediately
+        if (Visibility.isVisible()) {
+            if (prevProps.reportID !== this.props.reportID) {
+                this.unreadTimer = setTimeout(this.recordMaxAction, 3000);
+            } else if (!this.unreadTimer) {
```

Comment:
This controls the highlight/bolding on the LHN, basically.
> If the app is visible when the report actions view updates then we will check to see if the reportID has changed and if so set the last action as the last read but with a delay of 3 seconds
Correct. Effectively, this means that, when someone opens the chat (coming from a different chat) and there's new stuff, we wait 3 seconds and then update the last action (which unbolds the chat in the LHN)
The other part is there to prevent the LHN from highlighting every time a message comes in if the user is currently on the chat (which is currently happening). It immediately updates the lastAction instead of waiting 3 seconds.

Reply:
>The other part is there to prevent the LHN from highlighting every time a message comes in if the user is currently on the chat (which is currently happening). It immediately updates the lastAction instead of waiting 3 seconds.

Ok this took me a while to wrap my head around. I think because... if we want to immediately mark the report as read _in response to a new message coming in_ then we should do this _in reaction to_ the report updating with a new comment. And not simply whenever the component updates, but is not changing from one report to another. 

Is there some way to tell the difference between a generic update and the specific update where a new action was added to the report?

Reply:
We may be able to compare the number of actions? I'm unsure how consistently that'd work, though.